### DISCUSSION TITLE
Unable to execute `COPY FROM`, likely due to missing `ExecuteCopyFromDsl` trait

### DISCUSSION THREAD

**Discussion by lkral-navmatix** [weiznich/diesel_async#159] [Ideas]
### Versions

- **Rust:** 1.78.0
- **Diesel:** 2.2.0
- **Diesel_async:** 86cf7cd7a2ed9291a5aa79432b7c4fda98892845 from master branch
- **Database:** PG
- **Operating System** Ubuntu 20.04.05

### Feature Flags

- **diesel:** "chrono", "postgres"
- **diesel_async:** "postgres"

### What are you trying to accomplish?
Trying to use `COPY FROM` through diesel/diesel-async. I understand this version wasn't released yet but I was eager to test this new feature. 

### What is the expected output?
Being able to compile and execute:
```
    diesel::copy_from(example::table)
        .from_insertable(&data)
        .execute(db_conn)
        .await
```

### What is the actual output?
Attempting to use the `execute` from `diesel_async::RunQueryDsl` results in compilation errors:
`the trait ``QueryFragment<_>`` is not implemented for ...`
`the trait ``QueryId`` is not implemented for ...`

### Steps to reproduce

The code that triggers this is basically https://docs.diesel.rs/2.2.x/diesel/fn.copy_from.html#via-copyfromqueryfrom_insertable, just trying to use the diesel_async trait to run it with `AsyncPgConnection`.

It looks like diesel itself uses new trait `ExecuteCopyFromDsl` to execute this type of query, I can't currently see this in here so is likely missing. I can look into implementing it but not that familiar with insides of this crate so maybe someone more familiar could tackle this quicker.

Thank you

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/weiznich/diesel_async/issues) for similar possible closed issues.
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [x] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment]
This is expected behavior as this is a connection and not backend specific functionality. The connection provided by diesel-async just does not provide this feature yet. We are open to receive contributions for this feature, but otherwise this is something that's not planned to implement soon from my side.

**Comment by jr1221** [comment]
Is there any way to use `AsyncConnectionWrapper` to achieve the `ExecuteCopyFromDsl`?

**Comment by tylerjw** [comment]
Just chiming in to say this feature would be helpful for me.

**Reply by weiznich** [reply]
@tylerjw Your contributions are welcome as already pointed out above. Otherwise it's discouraged to post such comments here.

**Reply by tylerjw** [reply]
I'm sorry for the comment that adds nothing. I'll look into doing this.
